### ISSUE TITLE
[RFE] Subpackage for excluded files

### ISSUE THREAD

**Issue by voxik** [rpm-software-management/rpm#1448]
Probably silly idea, but in the context of #1442, could we have something like "-dev-null" subpackage, where we could reference all the files, which should not be packaged? Such package would not be created at the end, or it would not be composed into distribution. Maybe event the `%exclude` could be utilized to create file list for such subpackage?

**Comment by pmatilai** [comment]
I don't see rpm automatically creating such a package, but it's not like you *have to* ship every package that gets built. Happens in RHEL all the time, maybe Fedora should support the concept too? For example I could see packaging up tests being potentially useful for test-automation, but such packages hardly should be part of the distro itself.

Distro composition is outside rpm jurisdiction though.

**Comment by pmatilai** [comment]
Out of curiosity, what's the thing that rubygems systematically exclude?

**Comment by voxik** [comment]
rubygem- packages exclude cached versions of the .gem package itself (I don't want people to redistribute these, because they differ from the canonical packages available on rubygems.org), so every package contains `%exclude %{gem_cache}`. I don't think this one will be big deal to hide in `%gem_install` macro.

But I was thinking about this again working on Ruby. For example these files [[1]] should belong to the package. They are excluded just because there is external rubygem-racc and there is possible collision. If I remove such files in `%install` section, their exclusion will be less visible. You can note that I keep the files section quite explicit to have control what really goes into the packages. I work with the `%files` section all the time, while I touch the `%install` section quite rarely.

Sometimes I wish there was more control about files in `%{buildroot}` and RPM provided more insights, like "These are all files/directories which are in `%{buildroot}` and should be packaged. From this list, these files were already processed and there are sill few remaining". I know I could use filelists, but they on the other side hides a lot of stuff from the .spec file.


[1]: This package is  not the simplest one. I could scatter `rm` around, but still, some files logically belongs somewhere and they are not there just because of coll

**Comment by ffesti** [comment]
I agree that there should be more control over what files go where and the current means given to the packagers are not that great. Having more special code for creating sub packages is not something we want, though. We'd rather give the packager the means to do that in the spec - or the build scripts via dynamic spec files.

#2555 is somthing that's more likey to be implemented and may help here. Closing this ticket.
